src/template_parser.rs

```diff
@@ -521,6 +509,32 @@ fn parse_commit_or_change_id_method<'a, I: 'a>(
     Ok(property)
 }
 
+fn parse_shortest_id_prefix_method<'a, I: 'a>(
+    self_property: impl TemplateProperty<I, Output = ShortestIdPrefix> + 'a,
+    name: Pair<Rule>,
+    args_pair: Pair<Rule>,
+    _parse_keyword: &impl Fn(Pair<Rule>) -> TemplateParseResult<PropertyAndLabels<'a, I>>,
+) -> TemplateParseResult<Property<'a, I>> {
+    let property = match name.as_str() {
+        "with_brackets" => {
+            // TODO: If we had a map function, this could be expressed as a template
+            // like 'id.shortest() % (.prefix() if(.rest(), "[" .rest() "]"))'
```

Comment:
We could also have variables in the templating language, maybe. [No action needed]

Reply:
Maybe? But I would avoid procedural assignment.

Keywords can also be considered variables (with implicit `this_commit.`).

Reply:
I was thinking of just global constants and perhaps pure functions/methods.

For example, the commit template could then be composed of variables that are defined in other configs. So, you could separately change the definition `my_change_id` with your preferred style of change id one day, and  another day you could move around `my_change_id` within the commit template without worrying about the style of the change id.

It sounds like a big task, but considering the direction of where your refactoring of the templater is going, it might not be that far.

Reply:
That's basically a template alias. #1190

The problem for the moment is how to effectively write tests for alias expansion. Unlike revsets, templater doesn't have an intermediate structure that implements PartialEq/Debug.

Reply:
That point about tests is worth writing down in that bug. I'll put it there unless you tell me not to or put it there yourself (or unless I forget).